pkg/distributor/query.go

```diff
@@ -252,3 +252,10 @@ func sameSamples(a, b []ingester_client.Sample) bool {
 	}
 	return true
 }
+
+// LabelsToKeyString is used to form a string to be used as
+// the hashKey. Don't print, use l.String() for printing.
+func LabelsToKeyString(l labels.Labels) string {
+	b := make([]byte, 0, 1024)
```

Comment:
Using 1024 bytes slice seems like overkill. Here is a histogram of label lengths from single block with 949910 series:

```
66-155.3    █ 14629
155.3-244.6 ██████████████████████████████▏ 487865
244.6-333.9 ██████████████████ 292595
333.9-423.2 █▌ 23516
423.2-512.5 ▌ 6943
512.5-601.8 ███▊ 60915
601.8-691.1 ███▍ 54235
691.1-780.4 ▋ 8929
780.4-869.7 ▏ 277
869.7-959   ▏ 6

summary:
 p50=240, p90=562, p95=624, p99=690, min=66, max=959, avg=284.7459643545178
```

It shows that 90% of all series have labels shorter than 562 bytes.

(Alternatively: what about doing another pass through labels to get more accurate length to allocate?)

Reply:
Or using a pool?